src/com/codersbay/Main.java

```diff
@@ -279,47 +315,60 @@ public static int neighbors(boolean[][] habitat, int posX, int posY) {
         //X-1 Ymin | X Ymin | Xmax Ymin
         //X-1 Y    | X Y    | Xmax Y
         //X-1 Y-1  | X Y-1  | Xmax Y-1
-        if (posX == xMax && posY == yMax) {
+        if (posY == Max && posX == Max) {
 
-            for (int i = posX - 1; i <= posX; i++) {
-                for (int j = posY - 1; j <= posY; j++) {
+            for (int i = posY - 1; i <= posY; i++) {
+                for (int j = posX - 1; j <= posX; j++) {
                     if (habitat[i][j]) {
                         neighbor++;
                     }
                 }
             }
-            neighbor += habitat[posX - 1][0] ? 1 : 0;
-            neighbor += habitat[posX][0] ? 1 : 0;
-            neighbor += habitat[xMax][0] ? 1 : 0;
-            neighbor += habitat[xMax][posY] ? 1 : 0;
-            neighbor += habitat[xMax][posY - 1] ? 1 : 0;
+            //X-1 Ymin | X Ymin
+            neighbor += habitat[0][posX - 1] ? 1 : 0;
+            neighbor += habitat[0][posX] ? 1 : 0;
+
+            //Xmax Ymin
+            //Xmax Y
+            //Xmax Y-1
+            neighbor += habitat[0][Max] ? 1 : 0;
+            neighbor += habitat[posY][Max] ? 1 : 0;
+            neighbor += habitat[posY - 1][Max] ? 1 : 0;
 
             return neighbor;
         }
 
-        System.out.println("Error: neighbor [ " + posX + " ][ " + posY + "]");
-        return -1; //woot?!
+        System.out.println("Error: neighbor [ " + posY + " ][ " + posX + "]");
+        return -1; // in case of bullshit
     }
 
     public static boolean[][] cycleOfLife(boolean[][] habitat) {
```

Comment:
It's the circle of life 
https://open.spotify.com/track/0GrhBz0am9KFJ20MN9o6Lp?si=MzuWg33JRiqJMAdFWeEYOA